### PR TITLE
chore(tsconfig): align react, react-core, react-native, and ui tsconfig files

### PR DESCRIPTION
**Pull request by calebpollman**
#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/ui/tsconfig.json [hunk 1]
```diff
@@ -1,24 +1,16 @@
...
-    "declaration": true,
-    "outDir": "dist",
-    "baseUrl": ".",
-    "types": ["webpack-env", "jest"],
-    "paths": {
-      "@/*": ["src/*"]
-    },
-    "lib": ["esnext", "dom", "dom.iterable", "scripthost"]
+    "strict": false,
+    "target": "es6"
```
Comment (wlee221) [inline packages/ui/tsconfig.json:13]: Are we good to change target from `es5` to `es6` here?

File: docs/package.json [hunk 2]
```diff
@@ -78,6 +78,7 @@
     "react-markdown": "^8.0.3",
     "sitemap-urls": "^3.0.0",
     "ts-jest": "^27",
-    "ts-morph": "^15.1.0"
+    "ts-morph": "^15.1.0",
+    "ts-node": "^10.9.1"
```
[the remark this answers is not in the document]
Comment (dbanksdesign) [reply docs/package.json:82]: Looks like we removed ts-node in the UI package, do we really need it here? We were using ts-node to write scripts in TS I think and we changed to use something else... can we do that here (maybe out of scope)?

File: packages/ui/tsconfig.json (hunk 1 above)
Comment (calebpollman) [reply packages/ui/tsconfig.json:13]: IMO yes - `ui` is the only package that was using `es5` and as it is considered an internal package do not see any concerns with aligning it to the lowest common `target` of the platform packages which is `es6/es2015`

**Review by dbanksdesign — APPROVED** [review]
LGTM! kudos for the comments, made it really easy to follow. One non-blocking question/comment.

File: docs/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (calebpollman) [reply docs/package.json:82]: Docs failed to build without on the [initial CI](https://github.com/aws-amplify/amplify-ui/actions/runs/4288836933/jobs/7471547580) run. Think we can look in to outside of this PR of that's cool with you

**Review by reesscot — APPROVED** [review]
Don't be like me and merge before the release, but LGTM

**Review by wlee221 — APPROVED** [review]

**Pull request merged**